subscription.go

```diff
@@ -103,6 +103,12 @@ func (r *marathonClient) registerSubscription() error {
 	case EventsTransportCallback:
 		return r.registerCallbackSubscription()
 	case EventsTransportSSE:
+		if timeout := r.client.config.HTTPClient.Timeout; timeout > 0 {
```

Comment:
Do I understand correctly that this won't allow users to ever set the timeout even if they only wanted to have it for regular (non-SSE) requests?

Is it necessary to return the error to the user? How about simply disabling the critical timeout for SSE?